### PR TITLE
[Do not merge!] Pseudo PR for first release 

### PR DESCRIPTION
**Pull request by mashehu**
Do not merge! This is a PR of `dev` compared to the `TEMPLATE` branch for whole-pipeline reviewing purposes. Changes should be made to `dev` and this PR should not be merged into first-commit-for-pseudo-pr!

### CODE REVIEW

File: modules/local/add_columns/tests/tags.yml
Comment (mashehu) [inline modules/local/add_columns/tests/tags.yml:1]: I don't think this is needed anymore

File: modules/local/bam_chr_extract/tests/tags.yml
Comment (mashehu) [inline modules/local/bam_chr_extract/tests/tags.yml:1]: same here

File: nextflow.config [hunk 1]
```diff
@@ -9,20 +9,67 @@
 // Global default params, used in configs
 params {
 
-    // TODO nf-core: Specify your pipeline's command line flags
+    // steps
+    steps                        = null
```
Comment (mashehu) [inline nextflow.config:13]:
```suggestion
    steps                       = null
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -23,12 +23,34 @@
...
                 "outdir": {
                     "type": "string",
                     "format": "directory-path",
                     "description": "The output directory where the results will be saved. You have to use absolute paths to storage on Cloud infrastructure.",
                     "fa_icon": "fas fa-folder-open"
                 },
+                "rename_chr": {
+                    "type": "boolean",
+                    "description": "Should the panel vcf files be renamed to match the reference genome (e.g. 'chr1' -> '1')",
+                    "pattern": "true|false"
```
Comment (mashehu) [inline nextflow_schema.json:42]:
```suggestion
```
that is already checked with the type

File: nextflow_schema.json [hunk 3]
```diff
@@ -23,12 +23,34 @@
...
+                },
+                "max_chr_names": {
+                    "type": "integer",
+                    "description": "Maximum number of contigs name to print before resuming (i.e. show only subset and add '...' at the end).",
+                    "hidden": true,
+                    "default": 4
+                },
+                "remove_samples": {
+                    "type": "string",
+                    "description": "Comma-separated list of samples to remove from the reference panel. Useful for benchmarking purposes."
```
Comment (mashehu) [inline nextflow_schema.json:52]: add a pattern checking for a comma seperated list

File: nextflow_schema.json [hunk 4]
```diff
@@ -40,6 +62,141 @@
...
+                },
+                "normalize": {
+                    "description": "Should the reference panel be normalized",
+                    "type": "boolean",
+                    "pattern": "true|false"
+                },
+                "compute_freq": {
+                    "description": "Should the allele frequency for each variant (AC/AN fields necessary for Glimpse1 and the validation step) be computed using VCFFIXUP tool. This can be necessary if the fields are absent from the panel or if samples have been removed.",
+                    "type": "boolean",
+                    "pattern": "true|false"
```
Comment (mashehu) [inline nextflow_schema.json:130]:
```suggestion
```

File: nextflow_schema.json [hunk 5]
```diff
@@ -40,6 +62,141 @@
...
+            "title": "Imputation options",
+            "type": "object",
+            "description": "Arguments for the imputation steps",
+            "default": "",
+            "properties": {
+                "batch_size": {
+                    "type": "integer",
+                    "description": "Maximal number of individuals per batch for imputation.",
+                    "default": 100,
+                    "pattern": "^\\d+$"
```
Comment (mashehu) [inline nextflow_schema.json:148]:
```suggestion
```

File: nextflow_schema.json [hunk 6]
```diff
@@ -40,6 +62,141 @@
...
+                },
+                "bins": {
+                    "type": "string",
+                    "description": "User-defined allele count bins used for rsquared computations.",
+                    "default": "0 0.01 0.05 0.1 0.2 0.5",
+                    "pattern": "^(\\d+(\\.\\d+)? )+(\\d+(\\.\\d+)?)$"
+                },
+                "min_val_gl": {
+                    "type": "number",
+                    "description": "Minimum genotype likelihood probability P(G|R) in validation data. Set to zero to have no filter of if using gt-validation",
```
Comment (mashehu) [inline nextflow_schema.json:191]:
```suggestion
                    "description": "Minimum genotype likelihood probability P(G|R) in validation data. Set to zero to have no filter, if using gt-validation",
```

File: nextflow_schema.json [hunk 7]
```diff
@@ -40,6 +62,141 @@
...
+                    "default": 0.9,
+                    "pattern": "^\\d+(\\.\\d+)?$"
```
Comment (mashehu) [inline nextflow_schema.json:193]:
```suggestion
                    "pattern": "^\\d+(\\.\\d+)?$"
                    "minimum": 0
```

**Review by mashehu — COMMENTED** [review]
very nice job. the logic of when to choose what is a bit complicated sometimes, but you do a good deal helping people around that. 

Can't approve this PR because I created it, but will approve the release PR.

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -1,46 +1,107 @@
...
+      - master
   release:
     types: [published]
+  workflow_dispatch:
 
 env:
   NXF_ANSI_LOG: false
+  NXF_SINGULARITY_CACHEDIR: ${{ github.workspace }}/.singularity
+  NXF_SINGULARITY_LIBRARYDIR: ${{ github.workspace }}/.singularity
+  NFTEST_VER: "0.9.0"
```
[the remark this answers is not in the document]
Comment (atrigila) [reply .github/workflows/ci.yml:21]: thank you! we can use 0.9.2 that was launched yesterday and solves a few bugs :)

File: README.md [hunk 9]
```diff
@@ -5,86 +5,98 @@
...
 
-<!-- TODO nf-core: Include a figure that guides the user through the major workflow steps. Many nf-core
-     workflows use the "tube map" design for that. See https://nf-co.re/docs/contributing/design_guidelines#examples for examples.   -->
-<!-- TODO nf-core: Fill in short bullet-pointed list of the default steps in the pipeline -->
-
-1. Read QC ([`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/))
-2. Present QC for raw reads ([`MultiQC`](http://multiqc.info/))
+| Metro map                                                                       | Modes                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```
[the remark this answers is not in the document]
Comment (atrigila) [reply README.md:24]: thank you for your feedback, I agree, we will work on this

File: nextflow_schema.json (hunk 3 above)
Comment (atrigila) [reply nextflow_schema.json:52]: May I ask what you mean here? The idea is that if the user has a a reference haplotype panel that includes samples that he/she might want to remove, he could specify the names of the samples here. There is no pattern of those sample names.

File: docs/development.md
[the remark this answers is not in the document]
Comment (atrigila) [reply docs/development.md:1]: I added this to `CONTRIBUTING.md` but apparently it [breaks the linting ](https://github.com/nf-core/phaseimpute/actions/runs/11712828090/job/32624367995?pr=156) as it does not match the template

File: nextflow_schema.json (hunk 3 above)
Comment (atrigila) [reply nextflow_schema.json:52]: ohhh I see, very nice approach, I'll add that. thank you!

**Pull request closed without merging**